wolfcrypt/src/port/iotsafe/iotsafe.c

```diff
@@ -704,18 +776,25 @@ static int iotsafe_sign_hash(byte privkey_idx, uint16_t hash_algo,
         /* Terminate sign/sign session. */
         iotsafe_cmd_start(csim_cmd, IOTSAFE_CLASS, IOTSAFE_INS_SIGN_INIT, 1, 1);
         iotsafe_cmd_complete(csim_cmd);
-        if (expect_ok(csim_cmd, (word32)XSTRLEN(csim_cmd)) <= 0) {
+        if (expect_ok(csim_cmd, (word32)XSTRLEN(csim_cmd)) < 0) {
             ret = WC_HW_E;
         }
+        do {
+            ret = expect_ok("AT\r\n", 4);
+        } while (ret == 0);
```

Comment:
Same here... maybe add a timeout?

Reply:
done here, as well as in more similar do{}while loops that could lock up in case of hardware/applet failure